### PR TITLE
Add causal consistency

### PR DESCRIPTION
**Pull request by okdistribute**
There was a missing part of the docs that I think is very useful to understanding Ditto's CRDT guarantees around causal consistency.

### CODE REVIEW

**Review by russelldb — APPROVED** [review]
Thanks for adding this

File: docs/how-it-works/crdt.md [hunk 1]
```diff
@@ -86,3 +86,55 @@ Each Ditto document includes a version vector. The replication system uses the v
...
+
+Ditto's Causal Consistency ensures that if a new message is written after seeing some
+prior message, then the new message is not visible unless that prior message is
+also visible.
+
+To help differentiate Causal Consistency from stronger consistency models, imagine that Alice replies:
+
+    Alice: "Wonderful!"
+
+Causal Consistency allows that the two concurrent messages "Great news!"  from
```
Comment (hamchapman) [inline docs/how-it-works/crdt.md:125]:
```suggestion
Causal Consistency allows that the two concurrent messages "Great news!" from
```

File: docs/how-it-works/crdt.md [hunk 2]
```diff
@@ -86,3 +86,55 @@ Each Ditto document includes a version vector. The replication system uses the v
...
+within the same Ditto AppID. 
+
+To give an intuition about causal consistency,
+imagine the following scenario:
+
+On a social network Bob posts a message:
+
+    Bob: "I lost my cat"
+
+Then after sometime, he posts:
```
Comment (hamchapman) [inline docs/how-it-works/crdt.md:103]:
```suggestion
Then after some time, he posts:
```

**Review by hamchapman — APPROVED** [review]
Some unsolicited, drive-by review

**Comment by okdistribute** [comment]
Thanks all!

**Pull request merged**